### DISCUSSION TITLE
[Feature] Plans to Support MacOS

### DISCUSSION THREAD

**Discussion by IdolMenendez** [Voltstro-Studios/UnityWebBrowser#273] [Requests]
Hello，Thanks For The Project。
AnyPlan On MacOs Support？Or Just Pause the project

**Comment by Voltstro** [comment]
Supporting MacOS would be difficult for me or my team, as none of us own anything that runs MacOS.

### Intel MacOS

Support for Intel MacOS (10.x) could possibly be provided by using a virtual machine, its just that virutalizing MacOS is really slow from experience.
However I do plan to eventually support Intel MacOS.

### M1 ("Apple Silicon") ARM MacOS

Support for the new versions of MacOS (11.x) for the M1 chips is just something I can't do without physical hardware, and as far as I know virutualizing the new versions is impossible.
I could provide builds for it, but there would be no testing or support.
There is also the fact that the .NET 5, what the Browser Process side is built using, doesn't natively support the M1, however the upcoming [.NET 6 release](https://github.com/dotnet/runtime/issues/43313) will.

You could use Rosetta 2 emulation if I get a Intel MacOS build working, but once again no offical testing or support would be provided by using this way.

The only way to officaly support M1 MacOS is to have physical hardware, and well, I don't have the money for that.

**Comment by iMonZ** [comment]
> Supporting MacOS would be difficult for me or my team, as none of us own anything that runs MacOS.
> 
> ### Intel MacOS
> Support for Intel MacOS (10.x) could possibly be provided by using a virtual machine, its just that virutalizing MacOS is really slow from experience.
> However I do plan to eventually support Intel MacOS.
> 
> ### M1 ("Apple Silicon") ARM MacOS
> Support for the new versions of MacOS (11.x) for the M1 chips is just something I can't do without physical hardware, and as far as I know virutualizing the new versions is impossible.
> I could provide builds for it, but there would be no testing or support.
> There is also the fact that the .NET 5, what the Browser Process side is built using, doesn't natively support the M1, however the upcoming [.NET 6 release](https://github.com/dotnet/runtime/issues/43313) will.
> 
> You could use Rosetta 2 emulation if I get a Intel MacOS build working, but once again no offical testing or support would be provided by using this way.
> 
> The only way to officaly support M1 MacOS is to have physical hardware, and well, I don't have the money for that.

I can help you with an virtual machine or you can dual boot into macOS :)

**Comment by Voltstro** [comment]
MacOS support has been added in 2.2.0. However only the x64 version is tested, the arm64 builds I have no way of testing without real hardware.

The builds are also not code signed.
